src/crimson/osd/osd_operations/snaptrim_request.cc

```diff
@@ -31,6 +40,64 @@ seastar::future<> SnapTrimRequest::start()
 {
   logger().debug("{}", __func__);
   return seastar::now();
+
+  logger().debug("{}: start", *this);
+
+  IRef ref = this;
+  auto maybe_delay = seastar::now();
+  if (auto delay = 0; delay) {
```

Comment:
so `delay` is alway zero here?

Reply:
Removed! It turns this construct was accidentally hiding an issue with interruptor netsting (added an unit test for that).